### PR TITLE
parametric: re-enable config consistency tests for ruby

### PR DESCRIPTION
**Pull request by mabdinur**
## Motivation

Resolves the following errors: https://github.com/DataDog/system-tests/actions/runs/11689546728/job/32553275066 in the v2.5.0 release of the ruby tracer.

This PR updates the ruby tracer so it works with previous implementation of [Test_Config_UnifiedServiceTagging::test_specific_version](https://github.com/DataDog/system-tests/actions/runs/11689546728/job/32553275066)

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/parametric/test_config_consistency.py [hunk 1]
```diff
@@ -86,10 +86,9 @@ def test_default_config(self, library_env, test_agent, test_library):
 
     # Assert that iff a span has service name set by DD_SERVICE, it also gets the version specified in DD_VERSION
     @parametrize("library_env", [{"DD_SERVICE": "version_test", "DD_VERSION": "5.2.0"}])
-    @missing_feature(library="ruby")
     def test_specific_version(self, library_env, test_agent, test_library):
```
Comment (mabdinur) [inline tests/parametric/test_config_consistency.py:89]:
```suggestion
    @missing_feature(library < "ruby@2.7.0.dev")
    def test_specific_version(self, library_env, test_agent, test_library):
```

File: tests/parametric/test_config_consistency.py [hunk 2]
```diff
@@ -86,10 +86,9 @@ def test_default_config(self, library_env, test_agent, test_library):
...
         with test_library:
-            with test_library.start_span(name="s1") as s1:
+            with test_library.start_span(name="s1", service="version_test") as s1:
```
Comment (mabdinur) [inline tests/parametric/test_config_consistency.py:91]:
```suggestion
            with test_library.start_span(name="s1") as s1:
```

File: tests/parametric/test_config_consistency.py [hunk 3]
```diff
@@ -132,7 +132,7 @@ class Test_Config_TraceAgentURL:
     which would be unnecessarily complex.
     """
 
-    @missing_feature(library="ruby")
+    @bug(context.library == "golang", reason="APMAPI-390")
```
Comment (mabdinur) [inline tests/parametric/test_config_consistency.py:135]:
```suggestion
    @ missing_feature(library == "golang", reason="APMAPI-390")
```

File: tests/parametric/test_config_consistency.py [hunk 4]
```diff
@@ -89,7 +89,7 @@ def test_default_config(self, library_env, test_agent, test_library):
 
     # Assert that iff a span has service name set by DD_SERVICE, it also gets the version specified in DD_VERSION
     @parametrize("library_env", [{"DD_SERVICE": "version_test", "DD_VERSION": "5.2.0"}])
-    @missing_feature(library="ruby")
+    @missing_feature(library < "ruby@2.7.0.dev")
```
Comment (mabdinur) [inline tests/parametric/test_config_consistency.py:92]:
```suggestion
    @missing_feature(context.library < "ruby@2.7.0.dev")
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Failing in CI : https://github.com/DataDog/system-tests/actions/runs/11898131960/job/33154324222?pr=3381#step:5:270

**Comment by mabdinur** [comment]
> Failing in CI : https://github.com/DataDog/system-tests/actions/runs/11898131960/job/33154324222?pr=3381#step:5:270

[Fix](https://github.com/DataDog/dd-trace-rb/pull/4075) will be released in v2.8.0. I will update the version

File: tests/parametric/test_config_consistency.py [hunk 5]
```diff
@@ -89,7 +89,7 @@ def test_default_config(self, library_env, test_agent, test_library):
 
     # Assert that iff a span has service name set by DD_SERVICE, it also gets the version specified in DD_VERSION
     @parametrize("library_env", [{"DD_SERVICE": "version_test", "DD_VERSION": "5.2.0"}])
-    @missing_feature(library="ruby")
+    @missing_feature(context.library < "ruby@2.7.0.dev")
```
Comment (mabdinur) [inline tests/parametric/test_config_consistency.py:92]:
```suggestion
    @missing_feature(context.library < "ruby@2.8.0.dev")
```

**Comment by cbeauchesne** [comment]
You can declare it valid for `2.7.1-dev`, it will skip the test for `2.7.0`, and enable it for your main branch

**Comment by cbeauchesne** [comment]
Updating the branch to fix failure.

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**